frame/bridge/ethereum/issuing/src/lib.rs

```diff
@@ -0,0 +1,431 @@
+			.ok_or(<Error<T>>::UintCF)?;
+		let recipient = result.params[3]
+			.value
+			.clone()
+			.into_address()
+			.ok_or(<Error<T>>::AddressCF)?;
+		let fee = result.params[4]
+			.value
+			.clone()
+			.into_uint()
+			.ok_or(<Error<T>>::UintCF)?;
+
+		let input =
+			Abi::encode_cross_receive(dtoken_address.0.into(), recipient.0.into(), amount.0.into())
+				.map_err(|_| Error::<T>::InvalidMintEcoding)?;
+
+		log::trace!(target: "darwinia-issuing", "transfer fee will be delived to fee pallet {}", fee);
+		Ok(input)
+	}
+
+	pub fn mapped_token_address(
+		backing: EthereumAddress,
+		source: EthereumAddress,
+	) -> Result<H160, DispatchError> {
+		let factory_address = MappingFactoryAddress::get();
+		let bytes = Abi::encode_mapping_token(backing.0.into(), source.0.into())
+			.map_err(|_| Error::<T>::InvalidIssuingAccount)?;
+		let transaction =
+			Self::unsigned_transaction(U256::from(1), factory_address.0.into(), bytes);
+		let account = Self::dvm_account_id();
+		let mapped_address = T::DvmCaller::raw_call(account, transaction)
+			.map_err(|e| -> &'static str { e.into() })?;
+		if mapped_address.len() != 32 {
+			return Err(Error::<T>::InvalidAddressLen.into());
+		}
+		Ok(H160::from_slice(&mapped_address.as_slice()[12..]))
+	}
+
+	pub fn token_registed(
+		backing: EthereumAddress,
+		source: EthereumAddress,
+		target: EthereumAddress,
+	) -> DispatchResult {
+		let raw_event = RawEvent::TokenRegisted(0, backing, source, target);
+		let module_event: <T as Config>::Event = raw_event.clone().into();
+		let system_event: <T as frame_system::Config>::Event = module_event.into();
+		<BurnTokenEvents<T>>::append(system_event);
+		Self::deposit_event(raw_event);
+		T::EcdsaAuthorities::schedule_mmr_root(
+			(<frame_system::Pallet<T>>::block_number().saturated_into::<u32>() / 10 * 10 + 10)
+				.saturated_into(),
+		);
+		Ok(())
+	}
+
+	pub fn burn_token(
+		backing: EthereumAddress,
+		source: EthereumAddress,
+		recipient: EthereumAddress,
+		delegator: EthereumAddress,
+		amount: U256,
+	) -> DispatchResult {
+		let mapped_address = Self::mapped_token_address(backing, source).map_err(|e| {
+			log::trace!(target: "darwinia-issuing", "mapped token address error {:?} ", e);
+			e
+		})?;
+
+		let raw_event = RawEvent::BurnToken(
+			1,
+			backing,
+			recipient,
+			delegator,
+			source,
+			mapped_address.0.into(),
+			amount,
+		);
+		let module_event: <T as Config>::Event = raw_event.clone().into();
+		let system_event: <T as frame_system::Config>::Event = module_event.into();
+		<BurnTokenEvents<T>>::append(system_event);
+
+		Self::deposit_event(raw_event);
+		T::EcdsaAuthorities::schedule_mmr_root(
+			(<frame_system::Pallet<T>>::block_number().saturated_into::<u32>() / 10 * 10 + 10)
+				.saturated_into(),
+		);
+		Ok(())
+	}
+}
+
+impl<T: Config> ContractHandler for Module<T> {
+	/// handle
+	fn handle(address: H160, caller: H160, input: &[u8]) -> DispatchResult {
+		if MappingFactoryAddress::get() == caller.0.into() {
```

Comment:
if `MappingFactoryAddress::get() != caller.0.into()`, return dispatch error would be better?